### PR TITLE
44874 : Upgrade Pods

### PR DESCRIPTION
**Pull request by wbenabdessalem**
Solve the faced conflict after merge relevant to the upgrade pods as well as changes some line of code depend on the deprecated Swift language.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -26,8 +26,8 @@ target "eXo" do
 end
 
 target "share-extension" do
-    pod 'UICKeyChainStore', '~> 2.1'
-    pod 'HTMLKit', '~> 3.1'
+    pod 'UICKeyChainStore'
+    pod 'HTMLKit', '~> 3.1.0'
```
Comment (ahamdi) [inline Podfile:30]: We need to use a newer version of HTMLKit

File: Podfile.lock [hunk 2]
```diff
@@ -106,14 +106,15 @@ PODS:
...
+  - Crashlytics (~> 3.14.0)
   - Fabric
   - Firebase/Core
   - Firebase/Messaging
-  - HTMLKit (~> 3.1)
-  - SVProgressHUD (~> 2.2)
-  - SwiftyJSON (~> 4.0)
-  - UICKeyChainStore (~> 2.1)
+  - HTMLKit
+  - HTMLKit (~> 3.1.0)
```
Comment (ahamdi) [inline Podfile.lock:114]: We need to keep just one

File: eXo/Sources/Tool.swift [hunk 3]
```diff
@@ -78,36 +83,38 @@ class Tool {
...
-                        let version = (platformVersion as NSString).floatValue
-                        success(version)
+                do {
+                    let statusCode = (response as! HTTPURLResponse).statusCode
+                    if (statusCode >= 200  && statusCode < 300) {
+                        // Check platform version
+                        let json = try JSON(data: data!)
+                        if let platformVersion = json["platformVersion"].string {
+                            let version = (platformVersion as NSString).floatValue
+                            print(version)
```
Comment (ahamdi) [inline eXo/Sources/Tool.swift:102]: Do we still need this logging ?

File: eXo/eXoAppDelegate.swift [hunk 4]
```diff
@@ -182,13 +182,15 @@ class eXoAppDelegate: UIResponder, UIApplicationDelegate, MessagingDelegate, UNU
...
-        print("Push token reveived: \(fcmToken)")
+        if let _fcmToken = fcmToken {
+            print("Push token reveived: \(_fcmToken)")
+        }
     }
     
-    func messaging(_ messaging: Messaging, didReceive remoteMessage: MessagingRemoteMessage) {
-        print(remoteMessage.debugDescription)
+    func messaging(_ messaging: Messaging) {
+       // print(remoteMessage.debugDescription)
```
Comment (ahamdi) [inline eXo/eXoAppDelegate.swift:193]: If this function is no more needed , could we remove it ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: Podfile (hunk 1 above)
Comment (wbenabdessalem) [reply Podfile:30]: Yes the last changes :
pod 'UICKeyChainStore'
pod 'HTMLKit', '~> 3.1.0'

File: eXo/eXoAppDelegate.swift (hunk 4 above)
Comment (wbenabdessalem) [reply eXo/eXoAppDelegate.swift:193]: yes we can remove it

File: Podfile.lock (hunk 2 above)
Comment (wbenabdessalem) [reply Podfile.lock:114]: The second one : - HTMLKit (~> 3.1.0)

File: eXo/Sources/Tool.swift (hunk 3 above)
Comment (wbenabdessalem) [reply eXo/Sources/Tool.swift:102]: Yes we need that

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**